### PR TITLE
Refactor payment modal

### PR DESCRIPTION
**Pull request by nickpell**
refactorings / reducing code duplication that should make https://docs.google.com/a/dimagi.com/document/d/1b29UD47BfIJ_28QJQfl1tVm4BBoSdiiANbIlpB8VH-8/edit a whole lot easier

tested using cards from https://stripe.com/docs/testing

### CODE REVIEW

**Comment by nickpell** [comment]
@benrudolph

**Comment by biyeun** [comment]
Did this get QA'd?

**Comment by biyeun** [comment]
Since this is user facing and related to billing, I'd like to get this signed off by the QA team first. @christineawhite

**Comment by christineawhite** [comment]
@biyeun please make me a ticket when you are ready for me to take a look

**Comment by nickpell** [comment]
QA ticket: http://manage.dimagi.com/default.asp?162283

File: corehq/apps/domain/templates/domain/billing_statements.html [hunk 1]
```diff
@@ -11,12 +11,10 @@
...
         Stripe.setPublishableKey('{{ stripe_public_key }}');
 
-        var paymentHandler = new PaymentMethodHandler({
-            400: "{% trans "Your request was not formatted properly." %}",
-            403: "{% trans "Forbidden." %}",
-            404: "{% trans 'Page not found.' %}",
-            500: "{% blocktrans %}There was an error processing your request. We're working quickly to fix the issue. Please try again shortly.{% endblocktrans %}"
-        });
+        var paymentHandler = new PaymentMethodHandler(
+            {{ payment_error_messages|JSON }},
```
Comment (benrudolph) [inline corehq/apps/domain/templates/domain/billing_statements.html]: :+1:

**Comment by nickpell** [comment]
@biyeun QA passed, build failure looks unrelated

**Pull request merged**